checkov/terraform/checks/resource/base_resource_value_check.py

```diff
@@ -1,29 +1,92 @@
+    def _is_nesting_key(inspected_attributes, key):
+        """
+        Resolves whether a key is a subset of the inspected nesting attributes
+        :param inspected_attributes: list of nesting attributes
+        :param key: JSONPath key of an attribute
+        :return: True/False
+        """
+        return any([x in key for x in inspected_attributes])
+
     def scan_resource_conf(self, conf):
         inspected_key = self.get_inspected_key()
-        if dpath.util.search(conf, inspected_key) != {}:
-            value = dpath.util.get(conf, inspected_key)
-            if isinstance(value, bool):
-                value = [value]
-            if value[0] == self.get_expected_value():
+        expected_values = self.get_expected_values()
+        if dpath.search(conf, inspected_key) != {}:
+            if expected_values == [ANY_VALUE]:
+                # Key is found on the configuration - if it accepts any value, the check is PASSED
                 return CheckResult.PASSED
+            value = dpath.get(conf, inspected_key)
+            if isinstance(value, list) and len(value) == 1:
+                value = value[0]
+            if self._is_variable_dependant(value):
+                # If the tested attribute is variable-dependant, then result is PASSED
+                return CheckResult.PASSED
+            if value in expected_values:
+                return CheckResult.PASSED
+        else:
+            # Look for the configuration in a bottom-up fashion
+            inspected_attributes = self._filter_key_path(inspected_key)
+            for attribute in reversed(inspected_attributes):
+                for sub_key, sub_conf in dpath.search(conf, f'**/{attribute}', yielded=True):
+                    filtered_sub_key = self._filter_key_path(sub_key)
+                    if self._is_nesting_key(inspected_attributes, filtered_sub_key):
+                        if isinstance(sub_conf, list) and len(sub_conf) == 1:
+                            sub_conf = sub_conf[0]
+                        if sub_conf in self.get_expected_values():
+                            return CheckResult.PASSED
+                        if self._is_variable_dependant(sub_conf):
+                            # If the tested attribute is variable-dependant, then result is PASSED
+                            return CheckResult.PASSED
+
         return CheckResult.FAILED
 
     @abstractmethod
     def get_inspected_key(self):
-        raise NotImplemented()
+        """
+        :return: JSONPath syntax path of the checked attribute
+        """
+        raise NotImplementedError()
+
+    def get_expected_values(self):
+        """
+        Override the method with the list of acceptable values if the check has more than one possible expected value, given
+        the inspected key
+        :return: List of expected values, defaults to a list of the expected value
+        """
+        return [self.get_expected_value()]
 
     def get_expected_value(self):
```

Comment:
@tronxd Maybe we should rename to `get_default_passing_value`?

Reply:
please do not rename method names - for backward compatibility with eternal checks

Reply:
@nimrodkor didn't change because of the reason above